crates/typst-macros/src/symbols.rs

```diff
@@ -68,16 +94,50 @@ impl Parse for Kind {
 impl Parse for Variant {
     fn parse(input: ParseStream) -> Result<Self> {
         let mut name = String::new();
+        let handler = input.parse::<Handler>()?;
         if input.peek(syn::Ident::peek_any) {
             name.push_str(&input.call(Ident::parse_any)?.to_string());
             while input.peek(Token![.]) {
                 input.parse::<Token![.]>()?;
                 name.push('.');
                 name.push_str(&input.call(Ident::parse_any)?.to_string());
             }
+            if name == "self" && !handler.is_none() {
+                return Ok(Self {
+                    name: "".to_string(),
+                    c: LitChar::new('\0', Span::call_site()),
```

Comment:
If I use `floor` directly as a symbol / cast it to a string, do I get the null-terminator? That would be a bit of a problem.

Reply:
I tested the final pdf before. This symbol definition 

```rs
floor: [
    #[call(crate::math::floor)] self,
    l: '⌊',
    r: '⌋',
],
```

renders

```typ
$ A floor B $
```

into

![image](https://github.com/typst/typst/assets/69834864/c9f541ea-adf4-438a-8c42-969f993399f5)

Not sure if the `\0` does something unexpected.

Reply:
Okay, let's remove the `self` notation